File: Sesion 9/Ejercicio/Ejercicio3 S9/ejercicio3.py
import os;
class Empleado:
    def __init__(self, nombre, edad, DNI, puesto):
        self.nombre = nombre
        self.edad = edad
        self.DNI = DNI
        self.puesto = puesto

    def calcular_salario(self):
        pass

class Asalariados(Empleado):
    def __init__(self, nombre, edad, DNI, puesto, salario):
        super().__init__(nombre, edad, DNI, puesto)
        self.salario = salario

    def calcular_salario(self):
        return self.salario

class PorHora(Empleado):
    def __init__(self, nombre, edad, DNI, puesto, horas, pagoxhora):
        super().__init__(nombre, edad, DNI, puesto)
        self.horas = horas
        self.pagoxhora = pagoxhora

    def calcular_salario(self):
        return self.horas * self.pagoxhora

class Comisionistas(Empleado):
    def __init__(self, nombre, edad, DNI, puesto, ventas, comision):
        super().__init__(nombre, edad, DNI, puesto)
        self.ventas = ventas
        self.comision = comision

    def calcular_salario(self):
        return self.ventas * self.comision

def menu():
    print("====== MENU EMPLEADOS ======")
    print("1. Crear empleado Asalariado")
    print("2. Crear empleado PorHora")
    print("3. Crear empleado Comisionistas")
    print("4. Salir")
    return int(input("Ingresa tu opción: "))

def pausa_y_limpiar():
    input("Presiona Enter para continuar...")
    if os.name == 'nt':
        os.system('cls')  # Limpiar consola en Windows
    else:
        os.system('clear')  # Limpiar consola en Unix/Linux

def main():
    cont = 0
    array_empleados = []

    while True:
        opcion = menu()
        if opcion == 1:
            nombre = input("Ingrese el nombre del empleado Asalariado: ")
            edad = int(input("Ingrese la edad del empleado Asalariado: "))
            DNI = int(input("Ingrese el DNI del empleado Asalariado: "))
            puesto = input("Ingrese el puesto de trabajo del empleado Asalariado: ")
            salario = float(input("Ingrese el salario del empleado Asalariado: "))
            array_empleados[cont] = Asalariados(nombre, edad, DNI, puesto, salario)
            decision = input(f"¿Deseas saber el sueldo del empleado {nombre}? (SI/NO): ")
            if decision.upper() == "SI":
                print(f"El salario del empleado {nombre} es: {array_empleados[cont].calcular_salario()} soles.")
            cont += 1
            pausa_y_limpiar()
        elif opcion == 2:
            nombre = input("Ingrese el nombre del empleado PorHora: ")
            edad = int(input("Ingrese la edad del empleado PorHora: "))
            DNI = int(input("Ingrese el DNI del empleado PorHora: "))
            puesto = input("Ingrese el puesto de trabajo del empleado PorHora: ")
            horas = int(input("Ingrese las horas de trabajo del empleado PorHora: "))
            pagoxhora = float(input("Ingrese el pago por hora del empleado PorHora: "))
            array_empleados[cont] = PorHora(nombre, edad, DNI, puesto, horas, pagoxhora)
            decision = input(f"¿Deseas saber el sueldo del empleado {nombre}? (SI/NO): ")
            if decision.upper() == "SI":
                print(f"El salario del empleado {nombre} es: {array_empleados[cont].calcular_salario()} soles.")
            cont += 1
            pausa_y_limpiar()
        elif opcion == 3:
            nombre = input("Ingrese el nombre del empleado Comisionista: ")
            edad = int(input("Ingrese la edad del empleado Comisionista: "))
            DNI = int(input("Ingrese el DNI del empleado Comisionista: "))
            puesto = input("Ingrese el puesto de trabajo del empleado Comisionista: ")
            comision = float(input("Ingrese la tasa de comisión del empleado Comisionista: "))
            ventas = float(input("Ingrese el total de ventas del empleado Comisionista (EN DINERO): "))
            array_empleados[cont] = Comisionistas(nombre, edad, DNI, puesto, ventas, comision)
            decision = input(f"¿Deseas saber el sueldo del empleado {nombre}? (SI/NO): ")
            if decision.upper() == "SI":
                print(f"El salario del empleado {nombre} es: {array_empleados[cont].calcular_salario()} soles.")
            cont += 1
            pausa_y_limpiar()
        else:
            return


main()

File: Sesion 9/Ejercicio/Ejercicio3 S9/ejercicio3.cpp
#include <iostream>
using namespace std;

class Empleado {
    private:
        string nombre;
        int edad;
        int DNI;
        string puesto;
    public:
        Empleado(string n, int e, int D, string p) : nombre(n), edad(e), DNI(D), puesto(p) {}
        virtual double calcularSalario() = 0;
};

class Asalariados : public Empleado {
    private:
        double salario;
    public:
        Asalariados(string n, int e, int D, string p, double s) : Empleado(n, e, D, p), salario(s) {}
        double calcularSalario() override {
            return salario;
        }
};

class PorHora : public Empleado {
    private:
        int horas;
        double pagoxhora;
    public:
        PorHora(string n, int e, int D, string p, int h, double pxh) : Empleado(n, e, D, p), horas(h), pagoxhora(pxh) {}
        double calcularSalario() override {
            return horas * pagoxhora;
        }
};

class Comisionistas : public Empleado {
    private:
        double ventas;
        double comision;
    public:
        Comisionistas(string n, int e, int D, string p, double v, double c) : Empleado(n, e, D, p), ventas(v), comision(c) {}
        double calcularSalario() override {
            return ventas * comision;
        }
};

int menu() {
    int opc;
    cout << "====== MENU EMPLEADOS ======" << endl;
    cout << "1. Crear empleado Asalariado" << endl;
    cout << "2. Crear empleado PorHora" << endl;
    cout << "3. Crear empleado Comisionistas" << endl;
    cout << "4. Salir " << endl;
    cout << "Ingresa tu opción: ";
    cin >> opc;
    return opc;
}

void pausaYLimpiar() {
    cout << "Presiona Enter para continuar...";
    cin.ignore().get(); // Pausa
    #ifdef _WIN32
        system("CLS"); // Limpiar consola en Windows
    #else
        system("clear"); // Limpiar consola en Unix/Linux
    #endif
}

int main() {
    int opcion, edad, DNI, horas;
    string nombre, puesto, decision;
    double salario, pagoxhora, ventas, comision;
    int cont = 0;
    Empleado* array_empleados[100];
    
    while (true) {
        opcion = menu();
        if (opcion == 1) {
            cout << "Ingrese el nombre del empleado Asalariado: ";
            cin >> nombre;
            cout << "Ingrese la edad del empleado Asalariado: ";
            cin >> edad;
            cout << "Ingrese el DNI del empleado Asalariado: ";
            cin >> DNI;
            cout << "Ingrese el puesto de trabajo del empleado Asalariado: ";
            cin >> puesto;
            cout << "Ingrese el salario del empleado Asalariado: ";
            cin >> salario;
            array_empleados[cont] = new Asalariados(nombre, edad, DNI, puesto, salario);
            cout << "¿Deseas saber el sueldo del empleado " << nombre << "? (SI/NO): ";
            cin >> decision;
            if (decision == "SI") {
                cout << "El salario del empleado " << nombre << " es: " << array_empleados[cont]->calcularSalario() << " soles." << endl;
            }
            cont += 1;
            pausaYLimpiar();
        } else if (opcion == 2) {
            cout << "Ingrese el nombre del empleado PorHora: ";
            cin >> nombre;
            cout << "Ingrese la edad del empleado PorHora: ";
            cin >> edad;
            cout << "Ingrese el DNI del empleado PorHora: ";
            cin >> DNI;
            cout << "Ingrese el puesto de trabajo del empleado PorHora: ";
            cin >> puesto;
            cout << "Ingrese las horas de trabajo del empleado PorHora: ";
            cin >> horas;
            cout << "Ingrese el pago por hora del empleado PorHora: ";
            cin >> pagoxhora;
            array_empleados[cont] = new PorHora(nombre, edad, DNI, puesto, horas, pagoxhora);
            cout << "¿Deseas saber el sueldo del empleado " << nombre << "? (SI/NO): ";
            cin >> decision;
            if (decision == "SI") {
                cout << "El salario del empleado " << nombre << " es: " << array_empleados[cont]->calcularSalario() << " soles." << endl;
            }
            cont += 1;
            pausaYLimpiar();
        } else if (opcion == 3) {
            cout << "Ingrese el nombre del empleado Comisionista: ";
            cin >> nombre;
            cout << "Ingrese la edad del empleado Comisionista: ";
            cin >> edad;
            cout << "Ingrese el DNI del empleado Comisionista: ";
            cin >> DNI;
            cout << "Ingrese el puesto de trabajo del empleado Comisionista: ";
            cin >> puesto;
            cout << "Ingrese la tasa de comisión del empleado Comisionista: ";
            cin >> comision;
            cout << "Ingrese el total de ventas del empleado Comisionista (EN DINERO): ";
            cin >> ventas;
            array_empleados[cont] = new Comisionistas(nombre, edad, DNI, puesto, comision, ventas);
            cout << "¿Deseas saber el sueldo del empleado " << nombre << "? (SI/NO): ";
            cin >> decision;
            if (decision == "SI") {
                cout << "El salario del empleado " << nombre << " es: " << array_empleados[cont]->calcularSalario() << " soles." << endl;
            }
            cont += 1;
            pausaYLimpiar();
        } else {
            return 0;
        }
    }
}